x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java

```diff
@@ -53,6 +65,26 @@ protected AcknowledgedResponse read(StreamInput in) throws IOException {
     @Override
     protected void masterOperation(Task task, DeleteEnrichPolicyAction.Request request, ClusterState state,
                                    ActionListener<AcknowledgedResponse> listener) throws Exception {
+        List<PipelineConfiguration> pipelines = IngestService.getPipelines(state);
+        EnrichPolicy policy = EnrichStore.getPolicy(request.getName(), state);
+        List<String> pipelinesWithProcessors = new ArrayList<>();
+
+        for (PipelineConfiguration pipelineConfiguration : pipelines) {
+            List<AbstractEnrichProcessor> enrichProcessors =
+                ingestService.getProcessorsInPipeline(pipelineConfiguration.getId(), AbstractEnrichProcessor.class);
+            for (AbstractEnrichProcessor processor: enrichProcessors) {
+                if (processor.getPolicyName().equalsIgnoreCase(request.getName())) {
+                    pipelinesWithProcessors.add(pipelineConfiguration.getId());
+                }
+            }
+        }
+
+        if (pipelinesWithProcessors.isEmpty() == false) {
+            listener.onFailure(
+                new ElasticsearchStatusException("Could not delete policy [{}] because a pipeline is referencing it {}",
+                    RestStatus.CONFLICT, request.getName(), pipelinesWithProcessors));
```

Comment:
I think you need to collect and join (comma seperated) these pipelines, and likely adjust the corresponding test.

Reply:
Hey, this is already resolving it to a list (which is why there is no `[]` in the second arg in the string). I just for fun, added another processor to the test and got this error.

```
{"error":{"root_cause":[{"type":"status_exception","reason":"Could not delete policy [my_policy] because a pipeline is referencing it [my_pipeline, another_pipeline]"}],"type":"status
_exception","reason":"Could not delete policy [my_policy] because a pipeline is referencing it [my_pipeline, another_pipeline]"},"status":409}                                            

```